modules/Bio/EnsEMBL/Hive/RunnableDB/FastaFactory.pm

```diff
@@ -167,11 +173,13 @@ sub write_output {
     
     while (my $seq_object = $input_seqio->next_seq) {
 
+        next if ( $seq_object->id =~ /$seq_filter/ );
```

Comment:
In fact: can only do that if `$seq_filter` is defined

Reply:
Thats correct. Otherwise an empty regex would match all IDs and exclude all sequences. Fixed now.